src/python_extension.ts

```diff
@@ -114,10 +146,12 @@ export class PythonExtension {
 		pythonApi.settings.onDidChangeExecutionDetails(
 			(resource: Resource) => {
 				let newExecutionDetails = pythonApi.settings.getExecutionDetails(resource);
+				let folder = workspace.getWorkspaceFolder(resource);
+				let outermost = getOuterMostWorkspaceFolder(folder).uri;
```

Comment:
Why isn't this done in the updatePythonPath function so it doesn't need to be passed as argument in the callback? I'm sad it needs arguments as this is ugly :p But I don't see an easy way around it.

Reply:
I agree this is ugly, I got it to work that way and kept it at that since I'm not very comfortable with typescript callbacks, I'll have another quick look